core/src/main/java/org/apache/accumulo/core/clientImpl/lexicoder/ByteUtils.java

```diff
@@ -76,6 +76,9 @@
     for (int i = 0; i < in.length; i++) {
       if (in[i] == 0x01) {
         i++;
+        if (i >= in.length) {
```

Comment:
This should throw an exception.  This code is unescaping and always expects a character after 0x01.

Reply:
Interesting... it does make more sense that it expects and exception for bad data.